lib/Model/Verification.cpp

```diff
@@ -39,6 +39,33 @@ bool VerifyHelper::registerGlobalSymbol(const model::Identifier &Name,
   }
 }
 
+template<typename T>
+static std::string key(const T &Object) {
+  return getNameFromYAMLScalar(KeyedObjectTraits<T>::key(Object));
+}
+
+static std::string path(const model::Function &F) {
+  return "/Functions[" + key(F) + "]";
+}
+
+static std::string path(const model::DynamicFunction &F) {
+  return "/ImportedDynamicFunctions[" + key(F) + "]";
+}
+
+static std::string path(const model::TypeDefinition &T) {
+  return "/TypeDefinitions[" + key(T) + "]";
+}
+
+static std::string path(const model::EnumDefinition &D,
+                        const model::EnumEntry &Entry) {
+  return path(static_cast<const model::TypeDefinition &>(D))
+         + "/EnumDefinition.Entries[" + key(Entry) + "]";
+}
+
+static std::string path(const model::Segment &Segment) {
+  return "/Segments[" + key(Segment) + "]";
+}
```

Comment:
I would have expected these functions to be used somewhere else.
Were they not?

Reply:
I was surprised as well. But it looks like all the other places derive this "naturally" by serializing a reference: which I couldn't be happier about.

For a moment I _really_ wanted to convert this place too, but I managed to stop myself. I had enough fiddling with `TupleTrees` for one PR :)